### PR TITLE
RD-5776 Nodes widget refactoring

### PR DESCRIPTION
**Pull request by qooban**
## Description

This PR bases on https://github.com/cloudify-cosmo/cloudify-stage/pull/2452 and introduces the following changes to Nodes widget:

1. Add full translation support
2. Use lodash explicit imports or JS native functions
3. Use function components instead of class components
4. Code style improvements

When reviewing this PR I suggest to use "Hide whitespaces" option to make the diff smaller. 

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [ ] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
1. 2023-02-24: While latest Cloudify Manager image is broken I tested this PR locally - manual tests + Nodes widget Cypress spec - and I didn't find any issues.
2. 2023-02-27: [Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4568/) - only `nodes_spec`

## Documentation
N/A

### CODE REVIEW

File: widgets/nodes/src/NodesTable.tsx [hunk 1]
```diff
@@ -15,197 +20,171 @@ interface NodesTableProps {
...
-        const { data, toolbox, widget } = this.props;
-        const { error } = this.state;
-        const NO_DATA_MESSAGE = "There are no Nodes available. Probably there's no deployment created, yet.";
-        const { CopyToClipboardButton, DataTable, ErrorMessage, Icon, Popup } = Stage.Basic;
-
-        const { fieldsToShow } = widget.configuration;
-
-        return (
-            <div>
-                <ErrorMessage error={error} onDismiss={() => this.setState({ error: null })} autoHide />
```
Comment (qooban) [inline widgets/nodes/src/NodesTable.tsx:81]: `setState` for `error` state var was called only here, so effectively the error was never set.

File: widgets/nodes/src/NodesTable.tsx [hunk 2]
```diff
@@ -15,197 +20,171 @@ interface NodesTableProps {
...
-        const { toolbox } = this.props;
-        toolbox.refresh();
-    }
+export default function NodesTable({ data, toolbox, widget }: NodesTableProps) {
+    const refreshData = () => toolbox.refresh();
+    const fetchGridData: DataTableProps['fetchData'] = fetchParams => toolbox.refresh(fetchParams);
+    useEffect(() => {
+        toolbox.getEventBus().on('nodes:refresh', refreshData);
+        return () => toolbox.getEventBus().off('nodes:refresh', refreshData);
+    }, []);
```
Comment (kubama) [inline widgets/nodes/src/NodesTable.tsx:29]: Candidate for `useRefreshEvent`?

File: widgets/nodes/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export const widgetId = 'nodes';
+export const translateWidget = Stage.Utils.getT(`widgets.${widgetId}`);
```
Comment (kubama) [inline widgets/nodes/src/utils.ts:2]: Maybe `common.ts` would be a better suited name for this file?

File: widgets/nodes/src/NodesTable.tsx (hunk 2 above)
Comment (qooban) [reply widgets/nodes/src/NodesTable.tsx]: Good point, I forgot about that hook. Changed.

File: widgets/nodes/src/utils.ts (hunk 3 above)
Comment (qooban) [reply widgets/nodes/src/utils.ts]: Sure, changed.

**Review by kubama — APPROVED** [review]

File: widgets/nodes/src/NodeInstanceModal.tsx [hunk 4]
```diff
@@ -17,57 +19,60 @@ export default function NodeInstanceModal({ instance, onClose, open }: NodeInsta
...
                         >
-                            <DataTable.Column label="Target node" name="target" width="30%" />
-                            <DataTable.Column label="Relationship type" name="relationship" width="40%" />
-                            <DataTable.Column label="Source node" name="source" width="30%" />
+                            <DataTable.Column label={translate('relationships.targetNode')} name="target" width="30%" />
+                            <DataTable.Column label={translate('relationships.type')} name="relationship" width="40%" />
+                            <DataTable.Column label={translate('relationships.sourceNode')} name="source" width="30%" />
 
-                            {instance.relationships.map(r => {
+                            {instance.relationships.map(relationship => {
```
Comment (PatrycjaKubiczek) [inline widgets/nodes/src/NodeInstanceModal.tsx:47]: always nice to see such an improvement :) 🚀

**Review by PatrycjaKubiczek — APPROVED** [review]

**Pull request merged**